File: FamilyCalendarEventPlanner/src/FamilyCalendarEventPlanner.Api/Features/CalendarEvents/CalendarEventDto.cs
using FamilyCalendarEventPlanner.Core.Model.EventAggregate;
using FamilyCalendarEventPlanner.Core.Model.EventAggregate.Enums;

namespace FamilyCalendarEventPlanner.Api.Features.CalendarEvents;

public record CalendarEventDto
{
    public Guid EventId { get; init; }
    public Guid FamilyId { get; init; }
    public Guid CreatorId { get; init; }
    public string Title { get; init; } = string.Empty;
    public string Description { get; init; } = string.Empty;
    public DateTime StartTime { get; init; }
    public DateTime EndTime { get; init; }
    public string Location { get; init; } = string.Empty;
    public EventType EventType { get; init; }
    public RecurrencePatternDto RecurrencePattern { get; init; } = new();
    public EventStatus Status { get; init; }
}

public record RecurrencePatternDto
{
    public RecurrenceFrequency Frequency { get; init; }
    public int Interval { get; init; }
    public DateTime? EndDate { get; init; }
    public List<DayOfWeek> DaysOfWeek { get; init; } = new();
}

public static class CalendarEventExtensions
{
    public static CalendarEventDto ToDto(this CalendarEvent calendarEvent)
    {
        return new CalendarEventDto
        {
            EventId = calendarEvent.EventId,
            FamilyId = calendarEvent.FamilyId,
            CreatorId = calendarEvent.CreatorId,
            Title = calendarEvent.Title,
            Description = calendarEvent.Description,
            StartTime = calendarEvent.StartTime,
            EndTime = calendarEvent.EndTime,
            Location = calendarEvent.Location,
            EventType = calendarEvent.EventType,
            RecurrencePattern = new RecurrencePatternDto
            {
                Frequency = calendarEvent.RecurrencePattern.Frequency,
                Interval = calendarEvent.RecurrencePattern.Interval,
                EndDate = calendarEvent.RecurrencePattern.EndDate,
                DaysOfWeek = calendarEvent.RecurrencePattern.DaysOfWeek,
            },
            Status = calendarEvent.Status,
        };
    }
}

File: FamilyCalendarEventPlanner/src/FamilyCalendarEventPlanner.WebApp/projects/family-calendar-event-planner-admin-shared/src/lib/models/calendar-event-dto.ts
export type EventType = 'Appointment' | 'FamilyDinner' | 'Sports' | 'School' | 'Vacation' | 'Birthday' | 'Other';

export type EventStatus = 'Scheduled' | 'Cancelled' | 'Completed';

export type RecurrenceFrequency = 'None' | 'Daily' | 'Weekly' | 'Monthly' | 'Yearly';

export interface CalendarEventDto {
  eventId: string;
  familyId: string;
  creatorId: string;
  title: string;
  description: string;
  startTime: string;
  endTime: string;
  location: string;
  eventType: EventType;
  recurrencePattern: RecurrencePatternDto;
  status: EventStatus;
}

export interface RecurrencePatternDto {
  frequency: RecurrenceFrequency;
  interval: number;
  endDate?: string;
  daysOfWeek: string[];
}

export interface CreateEventCommand {
  familyId: string;
  creatorId: string;
  title: string;
  description?: string;
  startTime: string;
  endTime: string;
  location?: string;
  eventType: EventType;
  recurrencePattern?: RecurrencePatternDto;
}

export interface UpdateEventCommand {
  eventId: string;
  title?: string;
  description?: string;
  startTime?: string;
  endTime?: string;
  location?: string;
  eventType?: EventType;
  recurrencePattern?: RecurrencePatternDto;
}

export const EVENT_TYPES: { value: EventType; label: string }[] = [
  { value: 'Appointment', label: 'Appointment' },
  { value: 'FamilyDinner', label: 'Family Dinner' },
  { value: 'Sports', label: 'Sports' },
  { value: 'School', label: 'School' },
  { value: 'Vacation', label: 'Vacation' },
  { value: 'Birthday', label: 'Birthday' },
  { value: 'Other', label: 'Other' }
];

export const EVENT_STATUSES: { value: EventStatus; label: string }[] = [
  { value: 'Scheduled', label: 'Scheduled' },
  { value: 'Cancelled', label: 'Cancelled' },
  { value: 'Completed', label: 'Completed' }
];

export function getEventTypeLabel(eventType: EventType): string {
  return EVENT_TYPES.find(t => t.value === eventType)?.label || eventType;
}

export function getEventStatusLabel(status: EventStatus): string {
  return EVENT_STATUSES.find(s => s.value === status)?.label || status;
}
